research-software-directory/RSD-as-a-service
# 385 filter research domain

# Add filtering on research domain to projects page

Closes #385 

Changes proposed in this pull request:
*  User can filter for projects with a specific research domain from the project (item) page by clicking on the research domain tag. See example 1 screenshot
* User can filter projects for a specific research domain or combination of keywords and research domains on the project overview page. When filter is active the search input is applied within the filter selection. When filter is not applied the search term is also applied to keyword and research domain values of the project.
* Scaling image label is changed to: "Fit image into the area (recommended when using logos)" 
* Global search improved to include title+subtitle+keywords+research domain wild card search in software and projects while for organisation we search in name and the website.

How to test:
* `make start` to rebuild everything, when done use `docker-compose down`
* navigate to project page with research domain tags, you should be able to click on research domain which will open project overview page with the filter on selected research domain
* on project overview page you can select multiple filters and clear those
* try searching by keyword or research domain values, this should also work when filter is not applied.
* similar results should be achieved in global search

## Example 1 - research domain filter on project page
![image](https://user-images.githubusercontent.com/9204081/201137465-1070783e-c09d-48e1-a2db-e90e2f9ed29a.png)

## Example 2 - research domain filter on project overview page
![image](https://user-images.githubusercontent.com/9204081/201396453-3e899549-e242-414d-8908-23045e1c2026.png)

PR Checklist:
